waiter/src/waiter/scheduler.clj

```diff
@@ -47,6 +47,11 @@
   [& args]
   `(log/log "Scheduler" :debug nil (print-str ~@args)))
 
+(defmacro log-service-instance
+  "Log InstanceTracker-specific messages."
+  [instance event-type log-level key-seq]
+  `(log/log "InstanceTracker" ~log-level nil (utils/clj->json (assoc (select-keys ~instance ~key-seq) :timestamp (t/now) :event-type ~event-type))))
```

Comment:
The `key-seq` should be determined in this function. The caller should not be the one passing this list in.

Reply:
```
(let key-seq [:id :service-id :started-at :healthy? :health-check-status :flags :exit-code :host :port :extra-ports :log-directory :message]
  (log/log ...))
```

Would something like this suffice?

Reply:
We should just `def` the key-seq outside the macro and reference it inside.

Also, `:flags` needs to be flattened, otherwise it won't show correctly up in ELK.

Reply:
`:extra-ports` is also an array that needs to be flattened.

Reply:
> How would we request certain keys from the caller side in this case?

The caller shouldn't have to care. This function will extract whatever keys it feels are necessary to log the instance and event.

Reply:
I was thinking about `:extra-ports` again, and I think we should just log out the count. Since they are always sequential in both the Marathon and Kubernetes implementations, having port0 and the extra-ports count is sufficient.

Reply:
Done